include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -252,11 +235,45 @@ struct __subgroup_radix_sort
                                 {
                                     const uint16_t __r = __indices[__i];
                                     if (__r < __n)
-                                        __src[__r] = __keys[__i];
+                                    {
+                                        //move the values to source range and destroy the values
+                                        __src[__r] = ::std::move(__values.__v[__i]);
+                                        __values.__v[__i].~_ValT();
+                                    }
+                                }
+
+                                //destroy values in exchange buffer
+                                _ONEDPL_PRAGMA_UNROLL
+                                for (uint16_t __i = 0; __i < __block_size; ++__i)
+                                {
+                                    const uint16_t __idx = __wi * __block_size + __i;
+                                    if (__idx < __n)
+                                        __exchange_lacc[__idx].~_ValT();
                                 }
```

Comment:
(This could be left to a future exploration, I don't think the performance impact would be very large.)

Reply:
> Have you compared for these type of loops if it would be better performance-wise to calculate loop bounds using `__n`, rather than checking at each iteration? I'm not sure which is better because of unrolling vs extra branching and iterations.
> 
> For the data exchange loops, you should be able to make the loop bounds calc once, and share it.

I did it... Performance was worse.